torchbox/torchbox.com
# Update PhotoCollageBlock link configuration

[Link to Ticket](https://torchbox.monday.com/boards/1192293412/pulses/1393406114/posts/191019060)

### Description of Changes Made

This PR modifies the `PhotoCollageBlock`'s link setup so that both pages and arbitrary URLs can be specified.

### How to Test

Add/edit a photo collage block either on the homepage or on any service page

- there are two options for specifying the button link (internal link or external link). For either option, the correct link should be set on the button
- a button link requires accompanying button text (and vice-versa). A validation error should be raised when this condition is not satisfied
- the button link and button text are optional

### Screenshots

<details>
  <summary>Expand to see more</summary>

[![image](https://github.com/torchbox/torchbox.com/assets/7713776/ca9f672e-7772-4b01-9ed8-599e8d81d469)]([link_url](https://github.com/torchbox/torchbox.com/assets/7713776/ca9f672e-7772-4b01-9ed8-599e8d81d469))

[![image](https://github.com/torchbox/torchbox.com/assets/7713776/5c4baf00-c436-48a5-928c-ba2c616fad47)]([link_url](https://github.com/torchbox/torchbox.com/assets/7713776/5c4baf00-c436-48a5-928c-ba2c616fad47))

[![image](https://github.com/torchbox/torchbox.com/assets/7713776/b738944a-62ba-487d-bf0f-5894fa79f9d8)]([link_url](https://github.com/torchbox/torchbox.com/assets/7713776/b738944a-62ba-487d-bf0f-5894fa79f9d8))

</details>

### MR Checklist


#### Unit tests


#### Documentation


#### Browser testing

  - Latest version of Chrome on Linux
  - Latest version of Firefox on Linux

#### Data protection


#### Accessibility


#### Sustainability


#### Pattern library
